lib/bech32/test/Codec/Binary/Bech32Spec.hs

```diff
@@ -87,6 +96,116 @@ spec = do
             Bech32.encode hrp mempty
                 `shouldBe` Right (B8.pack "hrp1g9xj8m")
 
+    describe "Arbitrary ValidBech32String" $
+
+        it "Generation always produces a valid string that can be decoded." $
+            property $ \v ->
+                Bech32.decode (getValidBech32String v) `shouldBe`
+                    Right (humanReadablePart v, unencodedDataPart v)
+
+    describe "Decoding a corrupted string should fail" $ do
```

Comment:
did you have a look at these tests https://github.com/sipa/bech32/blob/master/ecc/javascript/segwit_addr_ecc.js#L27 ? (maybe they are not relevant)

Reply:
@akegalj wrote:

> did you have a look at these check bits https://github.com/sipa/bech32/blob/master/ecc/javascript/segwit_addr_ecc.js#L27 ? (maybe they are not relevant)

The Haskell translation of that particular function is at the following location:
`lib/bech32/src/Codec/Binary/Bech32/Internal.hs:392`

The test for that function is here:
`lib/bech32/test/Codec/Binary/Bech32Spec.hs:235`